### PR TITLE
[BACKLOG-40123] Adding two new extension points and updating TransMeta

### PR DESCRIPTION
**Pull request by pdesai16**
and JobMeta to call the new extension points.

### CODE REVIEW

**Review by niehusa — APPROVED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 50m 17s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core,engine
```
---
:ok_hand: All tests passed!
 
Tests run: **5373**, Failures: **0**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9231/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/core/extension/KettleExtensionPoint.java [hunk 1]
```diff
@@ -131,7 +131,9 @@ public enum KettleExtensionPoint {
     AfterDeleteRepositoryObject( "AfterDeleteRepositoryObject",
       "After an object has been deleted from the repository" ),
 
-    TransformationCreateNew( "TransformationCreateNew", "Create a New Empty Transformation in Spoon" );
+   TransformationCreateNew( "TransformationCreateNew", "Create a New Empty Transformation in Spoon" ),
+   TransSharedObjectsLoaded( "TransSharedObjectsLoaded", "After the shared objects are loaded in Transformations." ),
+   JobSharedObjectsLoaded( "JobSharedObjectsLoaded", "After the shared objects are loaded in Job." );
```
Comment (abryant-hv) [inline core/src/main/java/org/pentaho/di/core/extension/KettleExtensionPoint.java:136]: These are actually while the Meta is loaded, not the Job itself. There are other JobMeta/TransMeta ExtensionPoints. Maybe name these like those instead?

**Review by abryant-hv — APPROVED** [review]

File: core/src/main/java/org/pentaho/di/core/extension/KettleExtensionPoint.java (hunk 1 above)
Comment (abryant-hv) [reply core/src/main/java/org/pentaho/di/core/extension/KettleExtensionPoint.java:136]: Right, but these aren't being called when the Transformation or Job are being loaded, they're being called while the TransMeta and JobMeta are being loaded. I'm saying these should be named something like `TransformationMetaSharedObjectsLoaded` just like `TransformationMetaLoaded` (same with jobs).

Comment (pdesai16) [reply core/src/main/java/org/pentaho/di/core/extension/KettleExtensionPoint.java:136]: We can always add more context in the name but then it become too long.   I thought the two key things to add to the name is job/trans + when does it get called

Comment (abryant-hv) [reply core/src/main/java/org/pentaho/di/core/extension/KettleExtensionPoint.java:136]: "when does it get called" is not accurate here. All the "TransformationFoo" KettleExtensionPoints are called in Trans.java or when running a transformation. All the ones in TransMeta.java are "TransformationMetaFoo"

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 51m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core,engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.rest.RestTest.testCallEndpointWithGetVerb</b> (click to expand)</summary><p>

```
expected:<200> but was:<404>
```

</p></details>

  
Tests run: **5373**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9231/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 47m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core,engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.rest.RestTest.testCallEndpointWithGetVerb</b> (click to expand)</summary><p>

```
expected:<200> but was:<404>
```

</p></details>

  
Tests run: **5373**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9231/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 40m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core,engine
```
---
:ok_hand: All tests passed!
 
Tests run: **5384**, Failures: **0**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9231/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**